sycl/source/detail/config.cpp

```diff
@@ -175,8 +176,21 @@ getSyclDeviceTypeMap() {
   return SyclDeviceTypeMap;
 }
 
-// Array is used by SYCL_DEVICE_FILTER and SYCL_DEVICE_ALLOWLIST and
-// ONEAPI_DEVICE_SELECTOR
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
Can we just `#ifdef` a single line with `esimd_emulator` and use `auto` return type to unify those two versions?